### ISSUE TITLE
Low test accuracy for CIFAR100 dataset

### ISSUE THREAD

**Issue by joechan22** [google-research/vision_transformer#152]
I have conducted the experiment by using the Colab demonstration [https://colab.research.google.com/github/google-research/vision_transformer/blob/master/vit_jax.ipynb](url) with ViT-B_32 model and CIFAR100 dataset (instead of CIFAR10), however, the test accuracy is about 77% while it is expected to get about 90% accuracy. 
May i ask is it the setting issue or some other problem?

```
Parameters:
Model Type = ViT-B_32
dataset = CIFAR100 
total_steps = 500
warmup_steps = 50
decay_type = 'cosine'
grad_norm_clip = 1
```

**Comment by andsteing** [comment]
Where do you get the "90% accuracy" from?

For reproducing the numbers from the README (92.31% for the ViT-B/32 on CIFAR100 without dropout; reported as 91.97% in the [paper](https://arxiv.org/abs/2010.11929) in Table 5) you will need to run for 10k steps, as mentioned in table 7 and the config:

![image](https://user-images.githubusercontent.com/16050001/142880803-db792503-4ca7-4793-a7cb-753d49f30855.png)

https://github.com/google-research/vision_transformer/blob/39c905d2caf96a4306c9d78f05df36ddb3eb8ecb/vit_jax/configs/common.py#L94

You can also compare with the TensorBoard trace from the README:
https://tensorboard.dev/experiment/nwXQNjudRJW3dtQzhPZwwA/#scalars&regexInput=%5EViT-B_32%2Fcifar100%2Fdo_0.0&_smoothingWeight=0

Best, Andreas

**Comment by joechan22** [comment]
Hi Andreas, I am referring to the below table from the README of this repo. According to the result shown in the table, it is able to gain around 90% test accuracy with few total step size 500 and 1000 while I can only obtain around 77% accuracy with the same step size.

![image](https://user-images.githubusercontent.com/92969478/142971246-be036379-4c75-41c7-bba9-0df91d525025.png)

Thanks, Joe

**Comment by andsteing** [comment]
Hi Joe

Did you try with the ViT-B_16 model?

Best, Andreas

**Comment by joechan22** [comment]
Hi Andreas, 

yes, i have also tried to use ViT-B_16 with 500 total step size and 50 warm up step with around 78% accuracy. Thus, it is a bit confuse how to obtain 90% accuracy.

Thanks,
Joe

**Comment by andsteing** [comment]
How exactly did you run the ViT-B_16 model? Above you mention that you ran the ViT-B_32 model for 500 steps in https://colab.research.google.com/github/google-research/vision_transformer/blob/master/vit_jax.ipynb (but I don't think that the ViT-B_16  model will run with the same config, hence my question).

Best, Andreas

**Comment by joechan22** [comment]
Hi Andreas, 

I am just simply change the model name from "ViT-B_32" to "ViT-B_16" in the section of import while keeping all other parameters remain unchanged. Are there some hyperparameter that I need to amend for obtaining the expected result (~90% accuracy) under vit_jax.ipynb ?

Import Section
```
# Note: you can really choose any of the above, but this Colab has been tested
# with the models of below selection...
model_name = "ViT-B_16
```
other parameters
```
dataset = 'cifar100'
batch_size = 512
config = common_config.with_dataset(common_config.get_config(), dataset)
num_classes = input_pipeline.get_dataset_info(dataset, 'train')['num_classes']
config.batch = batch_size
config.pp.crop = 224
```
Final test accuracy after fine-tuning
```
INFO:absl:Load dataset info from /root/tensorflow_datasets/cifar100/3.0.2
100%|██████████| 19/19 [00:57<00:00,  3.01s/it]
DeviceArray(0.7843339, dtype=float32)
```

**Comment by andsteing** [comment]
The number we report in the paper are at 384 px, which is also the default in the config:

https://github.com/google-research/vision_transformer/blob/39c905d2caf96a4306c9d78f05df36ddb3eb8ecb/vit_jax/configs/common.py#L93-L99

The checked in configs are the ones to reproduce the numbers reported in the tables of the README as well, the Colab is just for illustration purposes. In order to speed up training there we decided to use larger batches and smaller images. You can try to comment out the line `config.pp.crop = 224` to use the 384 from the config, but you might need to adjust some of the other parameters to make it work.

Best, Andreas
